### ISSUE TITLE
Rustdoc does not escape GAT arguments when it fails to compute the href link for the trait

### ISSUE THREAD

**Issue by fmease** [rust-lang/rust#109488]
Given

```rs
pub type A = <S as Tr>::P<Option<i32>>; // `i32` instead of `Option<i32>` would work too but it's not as pronounced

pub struct S;
/*priv*/ trait Tr { type P<T>; }
impl Tr for S { type P<T> = (); }
```

Rustdoc does not escape the ASCII angle brackets for the generic argument list of the generic associated type projection `<S as Tr>::P` leading to the following butchered HTML output:

```
pub type A = <S as Tr>::P>; 
```

---

I've noticed this while working on rustdoc integration for `AliasKind::Inherent` in #109410. It's possible that the refactoring PR #109246 will fix this (need to check) or at least make it more pleasant to fix. Assigning myself for now since I'm gonna fix this otherwise.

@rustbot claim
@rustbot label T-rustdoc A-rustdoc-ui F-generic_associated_types

**Comment by jsha** [comment]
Copying my comment from #109246:

https://github.com/rust-lang/rust/blob/4c0f5008ce74563873cbd8574018dbe4906a5361/src/librustdoc/html/format.rs#L1158

`{:#}` is requesting alternate formatting (i.e. plaintext), but in this branch of things, alternate formatting hasn't been requested by the caller. So this is likely emitting `<` into an HTML context. If you change it to `{}` in your branch I'm guessing that will fix things.

**Comment by jsha** [comment]
Here's a live example: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_trait_selection/infer/trait.ObligationEmittingRelation.html#tymethod.register_predicates

As it turns out, this bug does appear to be what's crashing rustdoc in #109246. So I will wind up fixing it. But feel free to investigate and fix if you prefer not to wait.

**Comment by jsha** [comment]
This wound up being the fix: 236c18ff0c2f19ba9c74ebdd8a60aa005e2ff754

**Comment by fmease** [comment]
> This wound up being the fix: https://github.com/rust-lang/rust/commit/236c18ff0c2f19ba9c74ebdd8a60aa005e2ff754

On `master` at least I cannot reproduce your fix that involves adding `Escape(…)` in branch `Generic` (maybe that only works on your branch). In any case, the `assoc.args.print(cx)` lines in `format.rs` are definitely the culprit. I've already fixed it that way in #109410 but I've now also opened a separate PR.
